src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/BundleSlingInitialContentExtractor.java

```diff
@@ -0,0 +1,243 @@
+                                                                                        @NotNull File targetFile) throws UnsupportedEncodingException {
+        final String entryName = StringUtils.substringAfter( targetFile.getPath(), basePath + "/");
+        final PathEntry pathEntryValue = context.getPathEntryList().stream().filter(p -> entryName.startsWith( p.getPath())).findFirst().orElseThrow(NullPointerException::new);
+        final String target = pathEntryValue.getTarget();
+        // https://sling.apache.org/documentation/bundles/content-loading-jcr-contentloader.html#file-name-escaping
+        String repositoryPath = (target != null ? target : "/") + URLDecoder.decode(entryName.substring(pathEntryValue.getPath().length()), "UTF-8");
+        return new SlingInitialContentBundleEntryMetaData(targetFile, pathEntryValue, repositoryPath);
+    }
+
+
+
+    public void reset() {
+        parentFolderRepoInitHandler.reset();
+    }
+
+    public void addRepoinitExtension(@NotNull List<VaultPackageAssembler> assemblers, @NotNull FeaturesManager featureManager) throws IOException, ConverterException {
+        parentFolderRepoInitHandler.addRepoinitExtension(assemblers, featureManager);
+    }
+
+    protected void finalizePackageAssembly(@NotNull BundleSlingInitialContentExtractorContext context) throws IOException, ConverterException {
+        for (Map.Entry<PackageType, VaultPackageAssembler> entry : assemblerProvider.getPackageAssemblerEntrySet()) {
+            File packageFile = entry.getValue().createPackage();
+            ContentPackage2FeatureModelConverter converter = context.getConverter();
+            converter.processSubPackage(context.getPath() + "-" + entry.getKey(), context.getRunMode(), converter.open(packageFile), false);
+        }
+        assemblerProvider.clear();
+    }
+
+    private void safelyWriteOutputStream(@NotNull long compressedSize, 
+                                         @NotNull AtomicLong total, 
+                                         @NotNull byte[] data, 
+                                         @NotNull InputStream input, 
+                                         @NotNull OutputStream fos, 
+                                         boolean shouldClose) throws IOException {
+        int count;
+        BufferedOutputStream dest = new BufferedOutputStream(fos, BUFFER);
+        while (total.get() + BUFFER <= TOOBIG && (count = input.read(data, 0, BUFFER)) != -1) {
+            dest.write(data, 0, count);
+            total.addAndGet(count);
+
+            double compressionRatio = (double) count / compressedSize;
+            if(compressionRatio > THRESHOLD_RATIO) {
+                // ratio between compressed and uncompressed data is highly suspicious, looks like a Zip Bomb Attack
+                break;
+            }
+        }
+        dest.flush();
+
+        if(shouldClose){
```

Comment:
minor nitpicking: there should be a whitespace between if and the ( and before the {